src/orgwarden/__main__.py

```diff
@@ -1,17 +1,80 @@
+import sys
+from typing import Annotated
+from urllib.parse import urlparse
 import typer
-from .repo_crawler import fetch_org_repos
+from orgwarden.audit import audit_repository
+from orgwarden.repository import Repository
+from orgwarden.repo_crawler import fetch_org_repos
 
 app = typer.Typer()
 
 
 @app.command()
-def main(org_name: str) -> None:
+def list_repos(org_name: str) -> None:
+    """
+    Lists all public, non-forked repositories for the specified organization.
+    """
     repos = fetch_org_repos(org_name)
 
     print(f"~~~~~ Public repositories found for {org_name} ~~~~~")
     for repo in repos:
         print(f"{repo.org}/{repo.name} - {repo.url}")
 
 
+@app.command()
+def audit(
+    url: Annotated[
+        str, typer.Argument(help="The url for a GitHub repository or organization.")
+    ],
+) -> None:
+    """
+    If the provided <url> is a repository, runs RepoAuditor against the specified repository.
+    If the provided <url> is an organization, runs RepoAuditor against all of the organization's public, non-forked repositories.
+    """
+
+    def exit_for_invalid_url():
+        typer.echo(
+            typer.style(
+                f"Error: {url} is not a valid GitHub repository or organization.",
+                fg=typer.colors.RED,
+            )
+        )
+        typer.echo(
+            typer.style(
+                "Example: https://github.com/gt-tech-ai/OrgWarden",
+                fg=typer.colors.GREEN,
+            )
+        )
+        sys.exit(1)
+
+    # Parse url
+    parsed_url = urlparse(url)
+    if parsed_url.netloc != "github.com":
```

Comment:
Self-hosted github instances will be stored outside of github.com. For example, Georgia Tech's self-hosted github instance resides at `github.gatech.edu`.